### PR TITLE
GPU: Add auto aspect ratio and various fixes

### LINKED ISSUE #1139: 4:3 Aspect Ratio Identical to 1:1 PAR
**Description of the issue / bug**
The 4:3 aspect ratio option should be a little wider than 1:1 PAR, but currently they are the same (at least in the libretro core). I can achieve a proper 4:3 aspect by using the "All Borders" Crop Mode, which crops the vertical bars and stretches the image to 4:3. But ideally cropping options shouldn't change the aspect ratio.

**Steps to Reproduce**
Change the Aspect Ratio option to 4:3. It will have square pixels, just like the 1:1 PAR mode.

**Hardware/software**
Tested the libretro core in both Software and Vulkan renderer modes.

**Emulator version**
0.1-2433-g9089c973

### PR DESCRIPTION
**Pull request by ggrtk**
Auto mode width is roughly 2% slimmer than 4:3 mode due to using GPU ticks/lines for calculation. We could probably fudge the numbers a bit to make it closer to forcing 4:3, but analog specs aren't super precise anyways so I won't bother with it for the time being.

Should resolve #1139, #1138, #654.

### CODE REVIEW

**Comment by stenzek** [comment]
Would "Auto (4:3)" be a better name? Auto might be confused for being based on the window size.

**Comment by ggrtk** [comment]
How does "Auto (Native)" sound? Since it's not necessarily maintaining a 4:3 display aspect ratio depending on the user's crop settings.

**Comment by stenzek** [comment]
Hm, "Native" still sounds like it could be confused for the host. How about "Auto (Game)"? (open to other suggestions of course)

**Comment by DonelBueno** [comment]
Auto (Original)?

**Comment by ggrtk** [comment]
Maybe "Auto (Game Native)" or "Auto (Native to Game)"? Gets us pretty specific about it without being too wordy, I think.

**Comment by stenzek** [comment]
`Auto (Game Native)` works for me.

**Comment by ggrtk** [comment]
Done -- not sure how to expose this setting for the Android build, so if that could be handled separately it would be greatly appreciated. Thanks for the feedback!

**Comment by stenzek** [comment]
Great :) Will sort out Android and merge tomorrow with the other more significant changes.

**Comment by Awakened0** [comment]
With this pull, will only the Auto 4:3 aspect properly adjust when cropping is applied? Or will the other aspect options, particularly 1:1 PAR, also adjust to your cropping settings?

**Comment by ggrtk** [comment]
PAR 1:1 adjusts the aspect ratio for cropping adjustments because it's based on the visible resolution of the VRAM rectangle, which cropping affects. This is how it has been in standalone, and this behavior should be reflected in the libretro core as well.
